docs/develop/writingtests.rst

```diff
@@ -0,0 +1,79 @@
+.. _writingtests:
+
+Writing tests
+=============
+
+Tests in MicroPython are written in the path ``py/tests``:
+
+.. code-block:: bash
+   
+   .
+    ├── basics
+    ├── cmdline
+    ├── cpydiff
+    ├── esp32
+    ├── extmod
+    ├── feature_check
+    ├── float
+    ├── import
+    ├── inlineasm
+    ├── internal_bench
+    ├── io
+    ├── jni
+    ├── micropython
+    ├── misc
+    ├── multi_bluetooth
+    ├── multi_net
+    ├── net_hosted
+    ├── net_inet
+    ├── perf_bench
+    ├── pyb
+    ├── pybnative
+    ├── qemu-arm
+    ├── README
+    ├── run-internalbench.py
+    ├── run-multitests.py
+    ├── run-natmodtests.py
+    ├── run-perfbench.py
+    ├── run-tests
+    ├── run-tests-exp.py
+    ├── run-tests-exp.sh
+    ├── stress
+    ├── thread
+    ├── unicode
+    ├── unix
+    └── wipy
+
+There are subfolders maintained to categorize most tests. Add a test by creating a new file in one of the
+existing folders or in a new folder.
```

Comment:
I think it'll also work to make custom tests outside this tests folder (which would be recommended for a custom port).